.github/workflows/publish-to-pypi.yml

```diff
@@ -17,8 +17,14 @@ name: Publish to PyPI
 # Only run for pushes to the main branch and releases.
 on:
   push:
-    branches:
-      - main
+    branches: [ main ]
+    paths:
+      - 'pygmt/**'
+      - 'Makefile'
+      - 'MANIFEST.in'
+      - 'pyproject.toml'
+      - 'README.rst'
```

Comment:
Reminder to change this to 'README.md' once https://github.com/GenericMappingTools/pygmt/issues/2928 is done.